test/README.asciidoc

```diff
@@ -26,3 +29,48 @@ Usage
 To test, just type +run [test]+ in the +test+ directory.
 It will print each passing test.  If a test fails, a {unified-context-diff}[unified context diff]
 is printed showing the test’s expected output and the actual output.
+
+Details
+-------
+
++rc+ is optional
+and should contain a sequence of commands,
+_e.g._, +set-option+, +define-command+, +declare-option+.
++rc+ is sourced and evaluated before the +cmd+ key sequence is executed.
+
++cmd+ is required
+and should contain a key sequence that will edit the input buffer.
++cmd+ is executed after the +rc+ command sequence is sourced.
+
++in+ is optional
+and should contain the initial text loaded into the input buffer
+for editing by the +cmd+ key sequence.
+
++ui-in+ is optional
+and should contain the initial json-rpc commands loaded from the UI
+at startup to set a specific UI state.
```

Comment:
Not really, ui-in simulates the json-rpc commands sent by the ui, its not an "initial" set of commands in the sense that its the only set of ui commands coming in. For example, we could get rid of cmd and encode all the keystrokes directly as UI key commands inside the ui-in file, but that would be much more cumbersome to write.